poetry/config/config.py

```diff
@@ -140,6 +140,7 @@ def _get_normalizer(self, name: str) -> Callable:
             "virtualenvs.create",
             "virtualenvs.in-project",
             "virtualenvs.options.always-copy",
+            "experimental.new-installer",
             "installer.parallel",
         }:
```

Comment:
I decided to make sure that this list is kept correct in the future by having the tests generate a complete list of options that should be interpreted as booleans and validating each one. However it would be just as valid to generate the list here (or do it at the class level so that it happens once) and then hardcoding a few values to check in the tests.
Doing it this way would mean that we could hand pick test cases that we know cover a lot of conditions and check that the environment variable names are correct . For example

```python
@pytest.mark.parametrize(
    ("name", "env_var", "env_value", "value"),
    [
        ("installer.parallel", "POETRY_INSTALLER_PARALLEL", "true", True),
        ("installer.parallel", "POETRY_INSTALLER_PARALLEL", "false", False),
        ("virtualenvs.in-project", "POETRY_VIRTUALENVS_IN_PROJECT", "true", True),  # Using this poetry option makes sure that we replace both dots and dashes
        ("virtualenvs.in-project", "POETRY_VIRTUALENVS_IN_PROJECT", "false", False),
    ],
)
def test_config_get_from_environment_variable(...):
    ...
``` 

Let me know if you would prefer me to switch to using this approach instead.